### ISSUE TITLE
No charging indicator

### ISSUE THREAD

**Issue by hugoender** [odudex/krux#4]
Tried to add a charging indicator but failed. Here is what I tried:

In `power.py` file I added the following to use the `getBatteryChargeCurrent` method found in the MaixPy pmu:
```python
    def battery_charge_current(self):
        """Returns the charge current of the device's battery"""
        charge_current = int(self.pmu.getBatteryChargeCurrent())
        return charge_current
```

In ` __init__.py` added the following to the `draw_battery_indicator` method:
```python
battery_charge_current = self.ctx.power_manager.battery_charge_current()

# Draw a charging icon when device is charging
    if battery_charge_current > 0:
        self.ctx.display.draw_string(30, 0, ">", theme.go_color) 
```

I used `draw_string` just as a proof of concept. The final version would have a small lightning bolt or something next to the battery icon.

**Comment by odudex** [comment]
Great idea! It's hard to draw custom icons and drawings like a lightning bolt. But let's see what is possible.

**Comment by hugoender** [comment]
So that's the other question I had: is it possible to use unicode symbols? How was the Bitcoin symbol imported into the device? Maybe we can do something similar for lightning bolt.

**Comment by odudex** [comment]
Krux has a custom font format, in which the Bitcoin symbol was manually drawn and imported, it would be possible to do the same with a bolt. However, as the battery indicator doesn't change size in different devices like the fonts do, I believe it is better to implement the charging indicator with a custom drawing, not with a font glyph.

**Comment by hugoender** [comment]
How do you import a custom drawing?

**Comment by odudex** [comment]
Take a look at here:
firmware/font/README.md
and here:
firmware/MaixPy/projects/maixpy_m5stickv/compile/overrides/components/micropython/port/src/omv/img/font.c

**Comment by hugoender** [comment]
Thank you for those links.

Now I just need to figure out how to get the device to display regular text when it's charging and then I can worry about the custom drawing.

**Comment by odudex** [comment]
Solved with 5ef82caf29191ecc3b2602120e55a2c37dce7d7b
Battery status has a new design now, and its frame will get green (or "go" theme color) when charging. Note the icon and screen will refresh only after a button is pressed.
